test/evergreen.yml

```diff
@@ -199,7 +230,7 @@ functions:
       script: |
         set -o errexit
         # Confirm that the Python binary matches the version of that configured by CMake.
-        ${python_binary} ../test/evergreen/python_version_check.py -v -c ./CMakeCache.txt -s ${python_config_search_string}
+        ${python_binary|python3} ../test/evergreen/python_version_check.py -v -c ./CMakeCache.txt -s ${python_config_search_string}
```

Comment:
Do we still need `python_binary`? The pattern `${python_binary|python3}` is used multiple times in this file.

Reply:
yes unfortunately some tasks require their own python_binary to be set. E.g. windows, or tasks that use their own virtual env.